flipt-client-dart/lib/src/flipt_evaluation_client.dart

```diff
@@ -0,0 +1,231 @@
+
+typedef EvaluateBatchNative = Pointer<Utf8> Function(
+    Pointer<Void> engine, Pointer<Utf8> evaluationRequests);
+typedef EvaluateBatchDart = Pointer<Utf8> Function(
+    Pointer<Void> engine, Pointer<Utf8> evaluationRequests);
+
+typedef ListFlagsNative = Pointer<Utf8> Function(Pointer<Void> engine);
+typedef ListFlagsDart = Pointer<Utf8> Function(Pointer<Void> engine);
+
+typedef DestroyEngineNative = Void Function(Pointer<Void> engine);
+typedef DestroyEngineDart = void Function(Pointer<Void> engine);
+
+typedef DestroyStringNative = Void Function(Pointer<Utf8> str);
+typedef DestroyStringDart = void Function(Pointer<Utf8> str);
+
+class FliptEvaluationClient {
+  late Pointer<Void> _engine;
+  late DynamicLibrary _lib;
+
+  FliptEvaluationClient({String namespace = "default", Options? options}) {
+    _lib = DynamicLibrary.open(_getLibraryPath());
+    _initializeEngine(namespace, options);
+  }
+
+  String _getLibraryPath() {
+    final String libraryName;
+    final String platformDir;
+
+    switch (Platform.operatingSystem) {
+      case 'linux':
+        libraryName = 'libfliptengine.so';
+        platformDir =
+            Platform.version.contains('arm64') ? 'linux_arm64' : 'linux_x86_64';
+        break;
+      case 'macos':
+        libraryName = 'libfliptengine.dylib';
+        platformDir = Platform.version.contains('arm64')
+            ? 'darwin_arm64'
+            : 'darwin_x86_64';
+        break;
+      default:
+        throw UnsupportedError(
+            'Unsupported platform: ${Platform.operatingSystem}');
+    }
+
+    return path.join(
+      Directory.current.path,
+      'lib',
+      'src',
+      'ffi',
+      platformDir,
+      libraryName,
+    );
+  }
+
+  void _initializeEngine(String namespace, Options? options) {
+    final initializeEngine =
+        _lib.lookupFunction<InitializeEngineNative, InitializeEngineDart>(
+            'initialize_engine');
+
+    final namespaceUtf8 = namespace.toNativeUtf8();
+    final optsJson = jsonEncode(options?.toJson() ?? {});
+    final optsUtf8 = optsJson.toNativeUtf8();
+
+    try {
+      _engine = initializeEngine(namespaceUtf8, optsUtf8);
+    } finally {
+      calloc.free(namespaceUtf8);
+      calloc.free(optsUtf8);
+    }
+  }
+
+  Result<VariantEvaluationResponse> evaluateVariant(
+      {required String flagKey,
+      required String entityId,
+      required Map<String, dynamic> context}) {
+    final evaluateVariant =
+        _lib.lookupFunction<EvaluateVariantNative, EvaluateVariantDart>(
+            'evaluate_variant');
+
+    final request = EvaluationRequest(
+      flagKey: flagKey,
+      entityId: entityId,
+      context: context,
+    );
+
+    final requestJson = jsonEncode(request.toJson());
+    final requestUtf8 = requestJson.toNativeUtf8();
+
+    final resultPtr = evaluateVariant(_engine, requestUtf8);
+    final result = resultPtr.toDartString();
+    _destroyString(resultPtr);
```

Comment:
Just not critical thing but still it would be nice to have the same pattern between `evaluateBoolean`, `evaluateBatch` and `listFlags`. This section is out of try/catch block and `evaluateBoolean` has it inside. `listFlags` doesn't have try/catch block. Also I think we could call `calloc.free(requestUtf8);` much early.  Probably next to `_destroyString`.

Reply:
yeah i just removed the try / catch and called `calloc.free` right after `_destroyString`